File: kotlin-orders/kotlin-orders-dto/src/main/java/com/mvv/bank/orders/rest/entities/Amount.kt
package com.mvv.bank.orders.rest.entities

import java.math.BigDecimal


data class Amount (
    val value: BigDecimal,
    val currency: String,
) {
    override fun toString(): String = "$value $currency"

    companion object {
        // to have API similar to domain API
        @JvmStatic fun of(value: BigDecimal, currency: String) = Amount(value, currency)
    }
}

File: hello-world-samples/scala01/src/main/scala/com/mvv/bank/orders/rest/entities/Amount.scala
package com.mvv.bank.orders.rest.entities


case class Amount (
    value: BigDecimal,
    currency: String,
  ) :
  override def toString: String = s"$value $currency"

object Amount :
  def of(value: BigDecimal, currency: String): Amount = Amount(value, currency)
